src/tenderdash/initializeTenderdashNodeFactory.js

```diff
@@ -0,0 +1,113 @@
+const { WritableStream } = require('memory-streams');
+
+/**
+ *
+ * @param {DockerCompose} dockerCompose
+ * @param {Docker} docker
+ * @param {dockerPull} dockerPull
+ * @return {initializeTenderdashNode}
+ */
+function initializeTenderdashNodeFactory(dockerCompose, docker, dockerPull) {
+  /**
+   * @typedef {initializeTenderdashNode}
+   * @param {Config} config
+   * @return {Promise<Object>}
+   */
+  async function initializeTenderdashNode(config) {
+    if (await dockerCompose.isServiceRunning(config.toEnvs(), 'drive_tenderdash')) {
+      throw new Error('Can\'t initialize Tenderdash. Already running.');
+    }
+
+    const { COMPOSE_PROJECT_NAME: composeProjectName } = config.toEnvs();
+    const volumeName = 'drive_tenderdash_data';
+    const volumeNameFullName = `${composeProjectName}_${volumeName}`;
+
+    const volume = docker.getVolume(volumeNameFullName);
+
+    const isVolumeDefined = await volume.inspect()
+      .then(() => true)
+      .catch(() => false);
+
+    if (!isVolumeDefined) {
+      // Create volume with tenderdash data
+      await docker.createVolume({
+        Name: volumeNameFullName,
+        Labels: {
+          'com.docker.compose.project': composeProjectName,
+          'com.docker.compose.version': '1.27.4',
+          'com.docker.compose.volume': volumeName,
+        },
+      });
+
+      // Set tmuser ownership for the volume
+      await dockerPull('alpine:latest');
+
+      const writableStream = new WritableStream();
+
+      const command = [
+        'addgroup tmuser',
+        'adduser -S -G tmuser tmuser',
+        'chown -R tmuser:tmuser /tenderdash',
+      ].join('&&');
+
+      const [result] = await docker.run(
+        'alpine:latest',
+        ['sh', '-c', command],
+        writableStream,
+        {
+          HostConfig: {
+            AutoRemove: true,
+            Binds: [`${volumeNameFullName}:/tenderdash/data`],
```

Comment:
I think the problem you are trying to fix here is `/tendermint` is being created as root? It shouldn't be necessary to `chown` if you mount the volume at the mountpoint specified in the container, see discussion [here](https://github.com/dashevo/mn-bootstrap/pull/123#discussion_r478167920). Mounting at a deeper mountpoint will create parent dirs in the volume as root in your alpine container. Since the tenderdash container specifies its own volume already, the result looks like this (note the last volume):
```
"Mounts": [
            {
                "Type": "bind",
                "Source": "/home/strophy/.mn/evonet/tenderdash",
                "Destination": "/tenderdash/config",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "dash_masternode_evonet_drive_tenderdash_data",
                "Source": "/var/lib/docker/volumes/dash_masternode_evonet_drive_tenderdash_data/_data",
                "Destination": "/tenderdash/data",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "volume",
                "Name": "ee14dce8ede3fa522ca954433681ca85866a319121b3abc58a5ce03c28d21b8d",
                "Source": "/var/lib/docker/volumes/ee14dce8ede3fa522ca954433681ca85866a319121b3abc58a5ce03c28d21b8d/_data",
                "Destination": "/tenderdash",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
```

To avoid this hack, we could mount at `/tenderdash` instead, and either: 

- delete the `/tenderdash/config` files later after they are copied to the config object
- put another mount to catch the generated data in `/tenderdash/config` and drop it later

Reply:
Actually, the problem is that volume FS is under root, but we creating tenderdash home dir as `tmuser`. That's why we need to set the correct user for tenderdash volume first.

Reply:
I don't think this is true, we had exactly this problem with the `.dashcore` directory in the core image earlier this year. The tenderdash home dir already [exists](https://github.com/dashevo/tenderdash/blob/master/DOCKER/Dockerfile#L33) in the image with the correct owner. You are manually creating it again in a new volume with the wrong owner and then changing the owner, which is not necessary imo. If you mount it at the `/tenderdash` mountpoint, it will take the permissions already specified in the image. I have proven this locally as described above with a bind mount, but I'm getting an error "1" with no content from the writableStream when trying to implement it with a temp volume instead, and I don't know how to debug it...

I guess we can stick with your approach for now and create an issue to see if I can figure out a different way later?

Reply:
Fixed!